### PR TITLE
8309558: Create implementation of NSAccessibilityCheckBox protocol

### PR DESCRIPTION
**Pull request by azuev-java**
also
8309629: Create implementation of NSAccessibilityRadioButton protocol

Create implementation of NSAccessibilityCheckBox and NSAccessibilityRadioButton protocols
Add workaround for the wrong focus owner announcement with radio buttons
Add workaround for wrong magnifier text on buttons

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issues
 * [JDK-8309558](https://bugs.openjdk.org/browse/JDK-8309558): Create implementation of NSAccessibilityCheckBox protocol (**Enhancement** - P3)
 * [JDK-8309629](https://bugs.openjdk.org/browse/JDK-8309629): Create implementation of NSAccessibilityRadioButton protocol (**Enhancement** - P3)

### Reviewers
 * [Kevin Rushforth](https://openjdk.org/census#kcr) (@kevinrushforth - **Reviewer**)
 * [Ambarish Rapte](https://openjdk.org/census#arapte) (@arapte - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1184/head:pull/1184` \
`$ git checkout pull/1184`

Update a local copy of the PR: \
`$ git checkout pull/1184` \
`$ git pull https://git.openjdk.org/jfx.git pull/1184/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1184`

View PR using the GUI difftool: \
`$ git pr show -t 1184`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1184.diff">https://git.openjdk.org/jfx/pull/1184.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/1184#issuecomment-1646955053)

### CODE REVIEW

**Comment by kevinrushforth** [comment]
Reviewers @arapte @kevinrushforth

**Review by kevinrushforth — APPROVED** [review]
Code changes look good. I tested it on macOS 13.3.1 and VoiceOver output was as expected.

**Review by arapte — CHANGES_REQUESTED** [review]
Providing a minor comment for a warning.
Also noticed that, with this change tab are read as radio button. [ Tested with ensemble ]

Screenshot 2023-08-06 at 3 23 35 PM

File: modules/javafx.graphics/src/main/native-glass/mac/a11y/AccessibleBase.m [hunk 1]
```diff
@@ -162,6 +167,27 @@ - (void)clearParent
     parent = nil;
 }
 
+- (BOOL)isAccessibilityFocused
+{
+    jobject jresult = NULL;
+    GET_MAIN_JENV;
+    if (env == NULL) return NULL;
```
[the remark this answers is not in the document]
Comment (kevinrushforth) [reply modules/javafx.graphics/src/main/native-glass/mac/a11y/AccessibleBase.m:174]: Good catch. This should probably be changed to: `return NO;`

**Review by kevinrushforth — CHANGES_REQUESTED** [review]
> Also noticed that, with this change tab are read as radio button. [ Tested with ensemble ]

That seems like a bug.

File: modules/javafx.graphics/src/main/native-glass/mac/a11y/AccessibleBase.m (hunk 1 above)
[the remark this answers is not in the document]
Comment (azuev-java) [reply modules/javafx.graphics/src/main/native-glass/mac/a11y/AccessibleBase.m]: Fixed.

**Comment by azuev-java** [comment]
> Providing a minor comment for a warning. Also noticed that, with this change tab are read as radio button. [ Tested with ensemble ]
> 
> Screenshot 2023-08-06 at 3 23 35 PM

That seems to require the proper implementation of TAB_GROUP with the new API - mixing old and new API does not work here since in the existing code the accessible type reported is radio button but because parent encloses them in a tab group content it gets translated as tab item. Changing type to tab item without implementing tab group makes it loose the "Tab x out of y" notion so it is not the correct way to go. Commented out that assignment until tab group is implemented.

**Review by kevinrushforth — APPROVED** [review]

**Pull request closed without merging**